sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -310,8 +317,194 @@ public <T> Mono<CosmosItemResponse<T>> createItem(T item, CosmosItemRequestOptio
+                            mergeDiagnostics(response, cosmosException);
+                            return ModelBridgeInternal
+                                .createCosmosAsyncItemResponse(response, itemType, getItemDeserializer());
+                        })
+                        .single();
+
+                return readMono
+                    .onErrorMap(readThrowable -> {
+                        if (readThrowable instanceof CosmosException) {
+                            mergeDiagnostics(cosmosException, (CosmosException)readThrowable);
+                        }
+                        return cosmosException;
+                    })
+                    .flatMap(readResponse -> {
+                        if (readResponse.getStatusCode() == 200
+                        && itemResponseAccessor.hasTrackingId(readResponse, trackingId)) {
+                            return Mono.just(itemResponseAccessor.withRemappedStatusCode(
+                                readResponse,
+                                200,
+                                cosmosException.getRequestCharge(),
+                                this.isContentResponseOnWriteEffectivelyEnabled(options)));
+                        }
+
+                        return Mono.error(cosmosException);
+                    });
+            });
+    }
+
+    private <T> Mono<CosmosItemResponse<T>> createItemWithRetriesInternal(
+        T item, CosmosItemRequestOptions options, String trackingId) {
+
+        checkNotNull(trackingId, "Argument 'trackingId' must not be null.");
+
+        return createItemInternalCore(item, options, trackingId)
+            .onErrorResume(throwable -> {
+                Throwable error = throwable instanceof CompletionException ? throwable.getCause() : throwable;
+
+                if (!(error instanceof CosmosException)) {
+
+                    Exception nonCosmosException =
+                        error instanceof Exception ? (Exception) error : new RuntimeException(error);
+                    return Mono.error(nonCosmosException);
+                }
+
+                assert error instanceof CosmosException;
+                CosmosException cosmosException = (CosmosException) error;
+
+                if (cosmosException.getStatusCode() != HttpConstants.StatusCodes.CONFLICT) {
+                    return Mono.error(cosmosException);
+                }
+
+                InternalObjectNode internalObjectNode = InternalObjectNode.fromObjectToInternalObjectNode(item);
+                String itemId = internalObjectNode.getId();
+                RequestOptions requestOptions = ModelBridgeInternal.toRequestOptions(options);
+                @SuppressWarnings("unchecked")
+                Class<T> itemType = (Class<T>) item.getClass();
+
+                Mono<CosmosItemResponse<T>> readMono =
+                    this.getDatabase().getDocClientWrapper()
+                        .readDocument(getItemLink(itemId), requestOptions)
```

Comment:
the requestOption may not have the partitionKey info if the customer is calling through `container.createItem(Item, options)`, then the read request here can fail

Reply:
That is a typical Annie catch :-) Nice one - means my test cases are not sufficient yet

Reply:
Need to think about it

Reply:
Fixed